app/controllers/report_controller/saved_reports.rb

```diff
@@ -157,8 +157,13 @@ def set_saved_reports_condition(rep_id = nil)
 
     # Admin users can see all saved reports
     unless admin_user?
-      cond[0] << " AND miq_group_id=?"
-      cond.push(current_group_id)
+      cond[0] << " AND ("
```

Comment:
@lpichler I steered you wrong in asking you to put back cond.flatten at the end of the method. Seems like it forced you to do lots of string building.

Looking at your original commit I realize that was correct and clean because you needed this pattern for the condition:
`["miq_report_id IS NOT NULL AND miq_group_id IN (?)", [1,2, 3]]`
Flattening that would break the query.

You can put back what you originally had:
```
    unless admin_user?
      cond[0] << " AND miq_group_id IN (?)"
      cond.push(current_user.miq_groups.collect(&:id))
    end

    cond
```
That is much cleaner and correct.